tripy/nvtripy/frontend/tensor.py

```diff
@@ -175,12 +174,30 @@ def ndim(self):
     def device(self):
         return self.trace_tensor.device
 
-    # TODO (pranavm): Document this and `tolist()`? Figure out more ergonomic way to get scalar value.
-    def eval(self) -> runtime.MemRefValue:
+    def eval(self) -> "nvtripy.Tensor":
+        """
+        Immediately evaluates this tensor. By default, tensors are evaluated lazily.
+
+        Returns:
+            The evaluated tensor.
+
+        .. code-block:: python
+        :linenos:
+        :caption: eval
+
+            start = time.perf_counter()
+            tensor = tp.ones((3, 3))
+            tensor_construction = time.perf_counter()
+            tensor = tensor.eval()
```

Comment:
Just to make it clear that the tensor is being evaluated in place:
```suggestion
            tensor.eval()
```